client-java-core/src/main/java/io/featurehub/client/edge/EdgeRetryer.java

```diff
@@ -33,41 +38,39 @@ protected EdgeRetryer(int serverConnectTimeoutMs, int serverDisconnectRetryMs, i
     currentBackoffMultiplier = backoffMultiplier;
 
     executorService = makeExecutorService();
+
+    connectionStateActions.put(EdgeConnectionState.SUCCESS, (state, reConnector) -> currentBackoffMultiplier = backoffMultiplier);
+    connectionStateActions.put(EdgeConnectionState.API_KEY_NOT_FOUND, (state, reConnector) -> {
+      log.warn("[featurehub-sdk] terminal failure attempting to connect to Edge, API KEY does not exist.");
+      notFoundState = true;
+    });
+    connectionStateActions.put(SERVER_WAS_DISCONNECTED, (state, reConnector) -> reconnect(serverDisconnectRetryMs, true, state, reConnector));
+    connectionStateActions.put(SERVER_SAID_BYE, (state, reConnector) -> reconnect(serverByeReconnectMs, false, state, reConnector));
+    connectionStateActions.put(SERVER_CONNECT_TIMEOUT, (state, reConnector) -> reconnect(serverConnectTimeoutMs, true, state, reConnector));
   }
 
   // broken out for testability, can override with a mock pool
   protected ExecutorService makeExecutorService() {
     return Executors.newFixedThreadPool(1);
   }
 
-  public void edgeResult(EdgeConnectionState state, EdgeReconnector reconnector) {
+  public void edgeResult(EdgeConnectionState state, EdgeReconnector reConnector) {
     log.trace("[featurehub-sdk] retryer triggered {}", state);
-    if (!notFoundState && !executorService.isShutdown()) {
-      if (state == EdgeConnectionState.SUCCESS) {
-        currentBackoffMultiplier = backoffMultiplier;
-      } else if (state == EdgeConnectionState.API_KEY_NOT_FOUND) {
-        log.warn("[featurehub-sdk] terminal failure attempting to connect to Edge, API KEY does not exist.");
-        notFoundState = true;
-      } else if (state == EdgeConnectionState.SERVER_WAS_DISCONNECTED) {
-        executorService.submit(() -> {
-          backoff(serverDisconnectRetryMs, true);
-
-          reconnector.reconnect();
-        });
-      } else if (state == EdgeConnectionState.SERVER_SAID_BYE) {
-        executorService.submit(() -> {
-          backoff(serverByeReconnectMs, false);
-
-          reconnector.reconnect();
-        });
-      } else if (state == EdgeConnectionState.SERVER_CONNECT_TIMEOUT) {
-        executorService.submit(() -> {
-          backoff(serverConnectTimeoutMs, true);
-
-          reconnector.reconnect();
-        });
-      }
+    if (notFoundState || executorService.isShutdown()) {
+      return;
```

Comment:
I find this style a bit more of a cognitive load to follow, which is part of the reason I have kept it simple. I'm not entirely sure I like this...